src/DetailsView/components/requirement-view.tsx

```diff
@@ -77,6 +81,52 @@ export class RequirementView extends React.Component<RequirementViewProps> {
         };
     }
 
+    private renderNextRequirementButton(
+        requirement: Requirement,
+        assessment: Assessment,
+    ): JSX.Element {
+        const { mediumPassRequirementKeys, getProvider } = this.props.deps;
+        let nextRequirement: Requirement;
+        let currentTest: VisualizationType;
+        if (
+            this.props.shouldShowQuickAssessRequirementView &&
+            assessment.key !== AutomatedChecks.key
+        ) {
+            const requirementIndex = mediumPassRequirementKeys.findIndex(
+                r => r === requirement.key,
+            );
+            if (requirementIndex === mediumPassRequirementKeys.length - 1) {
+                //TODO: special go to assessment button
+                nextRequirement = null;
+                currentTest = this.props.assessmentNavState.selectedTestType;
+            } else {
+                const nextRequirementKey = mediumPassRequirementKeys[requirementIndex + 1];
+                const nextAssessment = getProvider().forRequirementKey(nextRequirementKey);
+                nextRequirement = nextAssessment.requirements.find(
+                    r => r.key === nextRequirementKey,
+                );
+                currentTest = nextAssessment.visualizationType;
+            }
+        } else {
+            const requirementIndex = assessment.requirements.findIndex(
```

Comment:
I think if we switch the statements and also return the next requirement button in the if block, we can make this a bit easier to read. Something like:

```
if (conditionForTypicalNextRequirementButton)
    return TypicalNextRequirementButton

return NewNextRequirementButton
```
```